### PR TITLE
fix(ui):  fix the errorPlaceholder visible when component is in loading state

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Fixes 

- I worked on fixing the errorPlaceholder visible when the component is in a loading state.
- Some code optimization 

https://user-images.githubusercontent.com/66266464/235653942-bbfc0030-78d8-4c31-86a4-fffbf6f1f7f5.mov

image

#
### Type of change:

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/TestSuitePipelineTab/TestSuitePipelineTab.component.tsx [hunk 1]
```diff
@@ -550,13 +553,44 @@ const TestSuitePipelineTab = () => {
...
   }
 
-  return !isAirflowAvailable ? (
-    <ErrorPlaceHolderIngestion />
-  ) : (
+  if (!isAirflowAvailable) {
+    return <ErrorPlaceHolderIngestion />;
+  }
+
+  if (isEmpty(testSuitePipelines) && !(isLoading || isFetchingStatus)) {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TestSuitePipelineTab/TestSuitePipelineTab.component.tsx:589]: && !(isLoading || isFetchingStatus)

This condition we already covers in above not sure how this helps, Can you check do we actually need this

File: openmetadata-ui/src/main/resources/ui/src/pages/database-details/index.tsx [hunk 2]
```diff
@@ -770,255 +770,273 @@ const DatabaseDetails: FunctionComponent = () => {
     setIsEditable(false);
   };
 
+  const databaseTable = useMemo(() => {
+    if (schemaDataLoading) {
+      return <Loader />;
+    } else if (!isEmpty(schemaData) && !schemaDataLoading) {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/database-details/index.tsx:776]:
```suggestion
    } else if (!isEmpty(schemaData)) {
```

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx [hunk 3]
```diff
@@ -863,11 +881,90 @@ const ServicePage: FunctionComponent = () => {
...
+                  : t('message.no-permission-for-action')
+              }>
+              <TestConnection
+                connectionType={serviceDetails?.serviceType ?? ''}
+                formData={connectionDetails as ConfigData}
+                isTestingDisabled={isTestingDisabled}
+                serviceCategory={serviceCategory as ServiceCategory}
+                serviceName={serviceDetails?.name}
+                // validation is not required as we have all the data available and not in edit mode
+                shouldValidateForm={false}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:945]: If we already have data validation will be returning true, We cannot skip validation in any mode

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:945]: We are using this TestConnection on the Service Detail page and we are directly hitting the API here, so don't need to validate here!
image

**Pull request merged**